packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -272,16 +291,37 @@ module AlgebraicCombination = {
       switch getInvalidOperationError(t1, t2, ~toPointSetFn, ~arithmeticOperation) {
       | Some(e) => Error(e)
       | None =>
-        switch chooseConvolutionOrMonteCarlo(arithmeticOperation, t1, t2) {
-        | MonteCarlo => runMonteCarlo(toSampleSetFn, arithmeticOperation, t1, t2)
-        | Convolution(convOp) =>
+        switch chooseConvolutionOrMonteCarlo(~strat=strategy, arithmeticOperation, t1, t2) {
+        | Ok(MonteCarlo) => runMonteCarlo(toSampleSetFn, arithmeticOperation, t1, t2)
+        | Ok(Convolution(convOp)) =>
           runConvolution(toPointSetFn, convOp, t1, t2)->E.R2.fmap(r => DistributionTypes.PointSet(
             r,
           ))
+        | Error(RequestedStrategyInvalidError) => Error(RequestedStrategyInvalidError)
+        | Error(err) => Error(err)
         }
       }
     }
   }
+
+  //  let run = (
+  //    ~mode: option<asMode>=?,
+  //    t1: t,
+  //    ~toPointSetFn: toPointSetFn,
+  //    ~toSampleSetFn: toSampleSetFn,
+  //    ~arithmeticOperation,
+  //    ~t2: t,
+  //  ): result<t, error> => {
+  //    let algebraicResult = run'(t1, ~toPointSetFn, ~toSampleSetFn, ~arithmeticOperation, ~t2)
+  //    switch (mode, algebraicResult) {
+  //    | (None, _)
+  //    | (Some(AsSymbolic), Ok(Symbolic(_)))
+  //    | (Some(AsMontecarlo), Ok(DistributionTypes.SampleSet(_)))
+  //    | (Some(AsConvolution), Ok(DistributionTypes.PointSet(_)))
+  //    | (Some(_), Error(_)) => algebraicResult
+  //    | (Some(_), Ok(_)) => Error(RequestedModeInvalidError)
+  //    }
+  //  }
```

Comment:
Comment deleted on next push.

Reply:
It's better to just delete the code for this PR. Save it somewhere else.